packages/teuchos/core/test/MemoryManagement/ArrayView_UnitTests.cpp

```diff
@@ -334,6 +350,7 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( ArrayView, danglingView_rcp_std_vector, T )
   }
 #ifdef TEUCHOS_DEBUG
   TEST_THROW(av.getRawPtr(), DanglingReferenceError);
+  TEST_THROW(av.data(), DanglingReferenceError);
```

Comment:
Okay, there is the dangling reference check

Reply:
But there are two `data()` functions (const and a non-const function).  Do both get tested?

Reply:
@bartlettroscoe I think only one gets tested.  If you like, I can delay this PR and push another commit that fixes this.

Reply:
> I think only one gets tested. If you like, I can delay this PR and push another commit that fixes this.

@mhoemmen, no need for delay.  I think this missing test did not show in the 100% coverage because all lines of code are covered but that does not technically guarantee full coverage of behavior. 

Thanks for adding these tests.

Cross fingers the PR tester will not fail again.